x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/service/rule_migrations_service.test.ts

```diff
@@ -0,0 +1,325 @@
+    });
+
+    it('should notify and not start migration if connectorId is missing', async () => {
+      (getMissingCapabilities as jest.Mock).mockReturnValue([]);
+      // Force connectorId to be missing
+      jest.spyOn(service.connectorIdStorage, 'get').mockReturnValue(undefined);
+
+      const result = await service.startRuleMigration('mig-1');
+      expect(mockNotifications.toasts.add).toHaveBeenCalled();
+      expect(result).toEqual({ started: false });
+    });
+
+    it('should start migration successfully when capabilities and connectorId are present', async () => {
+      (getMissingCapabilities as jest.Mock).mockReturnValue([]);
+      // Simulate a valid connector id and trace options
+      jest.spyOn(service.connectorIdStorage, 'get').mockReturnValue('connector-123');
+      jest.spyOn(service.traceOptionsStorage, 'get').mockReturnValue({
+        langSmithProject: 'proj',
+        langSmithApiKey: 'key',
+      } as TraceOptions);
+      (startRuleMigrationAPI as jest.Mock).mockResolvedValue({ started: true });
+
+      // Spy on startPolling to ensure it is called after starting the migration
+      const startPollingSpy = jest.spyOn(service, 'startPolling');
+      const result = await service.startRuleMigration(
+        'mig-1',
+        SiemMigrationRetryFilter.NOT_FULLY_TRANSLATED
+      );
+
+      expect(startRuleMigrationAPI).toHaveBeenCalledWith({
+        migrationId: 'mig-1',
+        connectorId: 'connector-123',
+        retry: SiemMigrationRetryFilter.NOT_FULLY_TRANSLATED,
+        langSmithOptions: { project_name: 'proj', api_key: 'key' },
+      });
+      expect(startPollingSpy).toHaveBeenCalled();
+      expect(result).toEqual({ started: true });
+    });
+  });
+
+  describe('getRuleMigrationStats', () => {
+    it('should return migration stats', async () => {
+      const stats = { id: 'mig-1', status: SiemMigrationTaskStatus.RUNNING };
+      (getRuleMigrationStats as jest.Mock).mockResolvedValue(stats);
+
+      const result = await service.getRuleMigrationStats('mig-1');
+      expect(getRuleMigrationStats).toHaveBeenCalledWith({ migrationId: 'mig-1' });
+      expect(result).toEqual(stats);
+    });
+  });
+
+  describe('getRuleMigrationsStats', () => {
+    it('should fetch and update latest stats', async () => {
+      const statsArray = [
+        { id: 'mig-1', status: SiemMigrationTaskStatus.RUNNING },
+        { id: 'mig-2', status: SiemMigrationTaskStatus.FINISHED },
+      ];
+      (getRuleMigrationsStatsAll as jest.Mock).mockResolvedValue(statsArray);
+
+      const result = await service.getRuleMigrationsStats();
+      expect(getRuleMigrationsStatsAll).toHaveBeenCalled();
+      expect(result).toHaveLength(2);
+      expect(result[0].number).toBe(1);
+      expect(result[1].number).toBe(2);
+
+      // Subscribe to the latestStats$ observable and check it was updated
+      const latestStats = await firstValueFrom(service.getLatestStats$());
```

Comment:
Consider asserting initial value of `null` and then updated value since that was the main change of this PR.

Reply:
I can't do it here since the `beforeEach` waits for the initialization. I added another test case here: https://github.com/elastic/kibana/pull/210188/commits/70c7717442f4cd87348c3809c90968c190b9f7dd

Reply:
makes sense. Thanks.